### PR TITLE
Fix the is operator on Ellipsis

### PR DESCRIPTION
**Pull request by eric-wieser**
This implements `a is b` on all opaque pointer types as `ptr_of(a) == ptr_of(b)`, which is exactly the meaning used by python anyway.

This implementation is copied from #3640.

### CODE REVIEW

**Comment by eric-wieser** [comment]
@stuartarchibald, should be ready for review

**Comment by stuartarchibald** [comment]
@eric-wieser thanks, have moved it into the queue.

**Comment by eric-wieser** [comment]
@sklam: How can I write that test using the `compile_isolated` style used by the rest of this file?

**Comment by stuartarchibald** [comment]
I don't think it's possible to use the `compile_isolated` style with something that cannot be unboxed.

**Comment by eric-wieser** [comment]
@stuartarchibald, where would you like me to add the suggested test?

**Comment by stuartarchibald** [comment]
> @stuartarchibald, where would you like me to add the suggested test?

Perhaps here: https://github.com/numba/numba/pull/5834/files#diff-fdfb3908ac2aefe7c395a313bf860a94R506 ?

**Comment by eric-wieser** [comment]
Added the test

File: numba/tests/test_operators.py [hunk 1]
```diff
@@ -491,6 +499,22 @@ def test_ne_scalar(self, flags=force_pyobj_flags):
...
         self.test_ne_scalar(flags=Noflags)
 
+    def test_is_ellipsis(self):
+        cres = compile_isolated(self.op.is_usecase, (types.ellipsis, types.ellipsis))
+        cfunc = cres.entry_point
+        self.assertTrue(cfunc(Ellipsis, Ellipsis))
+
+    def test_is_void_ptr(self):
+        # can't call this directly from python, as void cannot be unboxed
+        cfunc_void = njit((types.voidptr, types.voidptr))(self.op.is_usecase)
```
Comment (stuartarchibald) [inline numba/tests/test_operators.py:509]: Think `njit` needs importing or this needs to be `@jit(nopyhon=True)` will appropriate line wrap.

Comment (eric-wieser) [reply numba/tests/test_operators.py:509]: Any preference for which?

Comment (stuartarchibald) [reply numba/tests/test_operators.py:509]: Whichever you like :)

**Pull request merged**